text/0061-defrag-memos.md

```diff
@@ -0,0 +1,68 @@
+| 7 - 15     | Big-endian bytes of the total outlay amount, an unsigned 64-bit integer |
+| 15 - 23    | Big-endian bytes of the defragmentation transaction ID, an unsigned 64-bit integer |
+| 23 - 64    | Unused bytes |
+
+The fee amount is stored as a u56 in the first 7 bytes. This is done for consistency with RTH Destination memos. The next 8 bytes contain the total outlay of the defragmentation transaction. Lastly, the defragmentation ID is stored in 8 bytes as a u64.
+
+This memo will be written to the payload `TxOut` of a defragmentation transaction. Defragmentation transactions should not produce a change `TxOut`.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Accounts which have already performed defragmentation will either have no memos or sender/destination memos for their old defragmentation transactions. Though using an account with more than one application should be discouraged, not all developers will implement protections against it nor will all users heed this advice. If new applications do not handle legacy defragmentation transactions, they may inadvertently display them as normal transactions.
+
+This proposal will also require minor modifications to SDK clients. Since the addition of RTH, all transactions will generate a change `TxOut` regardless of whether or not there is change or if RTH memos are even being written at all. It is fairly trivial to make the required changes, however.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+One alternative to defragmentatino memos would be to continue using RTH for identifying defragmentation transactions. But this approach has a higher barrier of entry for client developers in terms of MobileCoin knowledge. It is much simpler to just check the memo type; and when it comes to sharing example code, defragmentation memos are much easier to demonstrate.
+
+Including a defragmentation ID may also not be necessary. Conversely, it may be better to use a larger value. A single u64 was chosen for simplicity and should be enough for one account. If clients do not need the ID field, they can also choose to ignore it.
+
+It may also be viable to implement this without removing change outputs from defragmentation transactions. The defragmentation memo could just be written to both outputs. Clients could differentiate the `TxOuts` still as the change amount would be 0. Change outputs are not necessary though for defragmentation, so it makes more sense to remove them. They will always have a value of 0 for defragmentation transactions and are not necessary for RTH purposes.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- Is a defragmentation ID necessary or valuable? Should it be larger than 64 bits?
+- Does this memo need any sort of validation? If client applications are hiding defragmentation transactions, users with custom clients could create 'invisible' transactions. This doesn't seem like a major issue, however.
```

Comment:
I think this can be mitigated by using some of the unused bytes to hold a signature that can only be created by the account owner.

Reply:
I think we should specify that the authentication works the same as for destination memos? that the memo belongs to the change subaddress? since its a self-payment